src/lib/concurrency/transaction_manager.hpp

```diff
@@ -48,11 +47,11 @@ class TransactionContext;
  * which represents the current global visibility of records.
  * The TransactionManager is thread-safe.
  */
-class TransactionManager : public Singleton<TransactionManager> {
+class TransactionManager : public Noncopyable {
   friend class TransactionManagerTest;
 
  public:
-  static void reset();
+  TransactionManager& operator=(TransactionManager&) = delete;
```

Comment:
What is this supposed to do? `operator=(TransactionManager&)` is not really a thing. `operator=(const TransactionManager&)` is. That one should be covered by Noncopyable. Am I missing something?